Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -0,0 +1,353 @@
+            let collection: CollectionV1
+            do {
+                // parse json
+                guard let decoded = try response.decodeBody(CollectionV1.self, using: self.decoder) else {
+                    throw Errors.invalidResponse("Invalid body")
+                }
+                collection = decoded
+            } catch {
+                return .failure(Errors.invalidJSON(error))
+            }
+
+            let packages = collection.packages.map { package -> PackageCollectionsModel.Collection.Package in
+                let versions = package.versions.compactMap { version -> PackageCollectionsModel.Collection.Package.Version? in
+                    // TODO: do we really want to ignore mapping errors, or should we return an error in these cases?
+                    guard let parsedVersion = TSCUtility.Version(string: version.version) else {
+                        return nil
+                    }
+                    guard let toolsVersion = ToolsVersion(string: version.toolsVersion) else {
+                        return nil
+                    }
+                    let targets = version.targets.map { PackageCollectionsModel.PackageTarget(name: $0.name, moduleName: $0.moduleName) }
+                    let products = version.products.compactMap { PackageCollectionsModel.PackageProduct(from: $0, packageTargets: targets) }
+                    let verifiedPlatforms: [PackageModel.Platform]? = version.verifiedPlatforms?.compactMap { PackageModel.Platform(from: $0) }
+                    let verifiedSwiftVersions = version.verifiedSwiftVersions?.compactMap { SwiftLanguageVersion(string: $0) }
+                    let license = version.license.flatMap { PackageCollectionsModel.License(from: $0) }
+                    return .init(version: parsedVersion,
+                                 packageName: version.packageName,
+                                 targets: targets,
+                                 products: products,
+                                 toolsVersion: toolsVersion,
+                                 verifiedPlatforms: verifiedPlatforms,
+                                 verifiedSwiftVersions: verifiedSwiftVersions,
+                                 license: license)
+                }
+                return .init(repository: RepositorySpecifier(url: package.url),
+                             summary: package.description,
+                             versions: versions,
+                             readmeURL: package.readmeURL.flatMap { Foundation.URL(string: $0) })
+            }
+            return .success(.init(source: source,
+                                  name: collection.name,
+                                  description: collection.description,
+                                  keywords: collection.keywords,
+                                  packages: packages,
+                                  createdAt: Date(), // TODO: is this correct?
```

Comment:
👀  @yim-lee

Reply:
Collection has `generatedAt`, which is what this should be.